vm/src/obj/objmappingproxy.rs

```diff
@@ -28,7 +28,8 @@ impl PyMappingProxy {
         if let Some(value) = objtype::class_get_attr(&self.class, key.as_str()) {
             return Ok(value);
         }
-        Err(vm.new_key_error(format!("Key not found: {}", key)))
+        let key_repr = vm.to_repr(key.as_object())?.to_string();
```

Comment:
`key` is already a string here. There is not need to convert it to a object and the calling `__repr__` on it.